src/main/java/com/google/firebase/messaging/AndroidNotification.java

```diff
@@ -309,6 +398,207 @@ public Builder setImage(String imageUrl) {
+     * persists even when the user clicks it.
+     *
+     * @param sticky The sticky flag
+     * @return This builder.
+     */
+    public Builder setSticky(boolean sticky) {
+      this.sticky = sticky;
+      return this;
+    }
+
+    /**
+     * Sets the time that the event in the notification occurred in milliseconds. Notifications
+     * in the panel are sorted by this time. The time will be formated in RFC3339 UTC "Zulu" 
+     * format, accurate to nanoseconds. Example: "2014-10-02T15:01:23.045123456Z". Note that 
+     * since the time is in milliseconds, the last section of the time representation always
+     * has 6 leading zeros.
+     *
+     * @param eventTimeInMillis The event time in milliseconds
+     * @return This builder.
+     */
+    public Builder setEventTimeInMillis(long eventTimeInMillis) {
+      this.eventTime = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS'Z'")
+          .format(new Date(eventTimeInMillis));
+      return this;
+    }
+
+    /**
+     * Sets whether or not this notification is relevant only to the current device. Some 
+     * notifications can be bridged to other devices for remote display, such as a Wear 
+     * OS watch. This hint can be set to recommend this notification not be bridged.
+     *
+     * @param localOnly The "local only" flag
+     * @return This builder.
+     */
+    public Builder setLocalOnly(boolean localOnly) {
+      this.localOnly = localOnly;
+      return this;
+    }
+
+    /**
+     * Sets the relative priority for this notification. Priority is an indication of how much of 
+     * the user's attention should be consumed by this notification. Low-priority notifications 
+     * may be hidden from the user in certain situations, while the user might be interrupted 
+     * for a higher-priority notification. The effect of setting the same priorities may differ 
+     * slightly on different platforms.
+     *
+     * @param priority The priority value, one of the values in {MIN, LOW, DEFAULT, HIGH, MAX}
+     * @return This builder.
+     */
+    public Builder setPriority(Priority priority) {
+      this.priority = priority;
+      return this;
+    }
+
+    /**
+     * Adds a vibration timing in milliseconds in the array to use. The first value in the 
+     * array indicates the duration to wait before turning the vibrator on. The next value 
+     * indicates the Duration to keep the vibrator on. Subsequent values alternate between 
+     * duration to turn the vibrator off and to turn the vibrator on. If vibrate_timings 
+     * is set and default_vibrate_timings is set to true, the default value is used instead
+     * of the user-specified vibrate_timings.
+     * A duration in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s".
+     *
+     * @param vibrateTimingInMillis vibration time in milliseconds.
+     * @return This builder.
+     */
+    public Builder addVibrateTimingsInMillis(long vibrateTimingInMillis) {
```

Comment:
Lets just expose one setter method for vibrateTimings:

```
public Builder setVibrateTimingsInMillis(long[] timings) {
   // Create a new List and add the array contents
}
```

Reply:
Done.